src/pages/candidates/Candidates.jsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState } from 'react';
+import { Box, FormControl, Select, MenuItem, InputLabel } from '@mui/material';
+import { AgGridColumn, AgGridReact } from 'ag-grid-react';
+import 'ag-grid-enterprise';
+import 'ag-grid-community/dist/styles/ag-grid.css';
+import 'ag-grid-community/dist/styles/ag-theme-alpine.css';
+import listOfCandidates from '../../mocks/listOfCandidates.json';
+import { tableFields } from '../../constants';
+import { getFieldLabel } from '../../utils';
+
+const Candidates = () => {
+  const [gridApi, setGridApi] = useState(null);
+  const onFirstDataRendered = (params) => {
+    const colIds = params.columnApi.getAllColumns().map((c) => c.colId);
+    params.columnApi.autoSizeColumns(colIds);
+  };
+
+  const onGridReady = (params) => {
+    setGridApi(params.api);
+
+    listOfCandidates.forEach((data, index) => {
+      const list = data;
+      list.id = index;
+    });
+    params.api.setRowData(listOfCandidates);
+  };
+
+  const onSortChanged = () => {
+    const sortModel = gridApi.getSortModel();
+    const sortActive = sortModel && sortModel.length > 0;
+    gridApi.setSuppressRowDrag(sortActive);
+  };
+
+  const onFilterChanged = () => {
+    const filterActive = gridApi.isAnyFilterPresent();
+    gridApi.setSuppressRowDrag(filterActive);
+  };
+
+  const onRowDragMove = (event) => {
+    function moveInArray(arr, fromIndex, toIndex) {
```

Comment:
I don't like the name function. 

Also you can move 'moveInArray'  outside 'onRowDragMove'. Add return